### PR TITLE
Fix #8043: Fix showcase lazy datatable example

### LINKED ISSUE #8043: Showcase: Lazy Datatable example is broken
**Describe the defect**
Lazy Datatable example in Showcase is broken.

**Reproducer**

1. Go here: http://primefaces.org/showcase/ui/data/datatable/lazy.xhtml

2. Sort the Country field.

```
java.lang.RuntimeException: java.lang.NoSuchFieldException: country
        at org.primefaces.showcase.view.data.datatable.LazySorter.compare(LazySorter.java:51)
        at org.primefaces.showcase.view.data.datatable.LazySorter.compare(LazySorter.java:30)
        at org.apache.commons.collections4.comparators.ComparatorChain.compare(ComparatorChain.java:281)
        at java.base/java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
        at java.base/java.util.TimSort.sort(TimSort.java:220)
        at java.base/java.util.Arrays.sort(Arrays.java:1515)
        at java.base/java.util.ArrayList.sort(ArrayList.java:1750)
        at org.primefaces.showcase.view.data.datatable.LazyCustomerDataModel.load(LazyCustomerDataModel.java:93)
        at org.primefaces.component.datatable.DataTable.loadLazyData(DataTable.java:488)
        at org.primefaces.component.datatable.feature.SortFeature.encode(SortFeature.java:112)
        at org.primefaces.component.datatable.DataTableRenderer.encodeEnd(DataTableRenderer.java:88)
        at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:675)
        at javax.faces.component.UIData.encodeEnd(UIData.java:1835)

```

### PR DESCRIPTION
**Pull request by melloware**
Couldn't access the fields using Reflection

### CODE REVIEW

**Comment by tandraschko** [comment]
better use PropertyDescriptor and getter/setter

**Comment by melloware** [comment]
Yeah I was wondering that for Java 17 compatibility going forward.

**Comment by melloware** [comment]
Fixed with PropertyDescriptor!  Good advice!

File: primefaces-showcase/src/main/java/org/primefaces/showcase/util/ShowcaseUtil.java [hunk 1]
```diff
@@ -64,6 +66,11 @@ private ShowcaseUtil() {
         return files;
     }
 
+    public static final Object getReflectionValue(Object o, String field)
```
Comment (tandraschko) [inline primefaces-showcase/src/main/java/org/primefaces/showcase/util/ShowcaseUtil.java]: i would use `getPropertyValue` or if you like `getPropertyValueViaReflection`

**Pull request merged**